### PR TITLE
Make Firebase Storage compile and work in independent watch app

### PR DESCRIPTION
**Pull request by chliangGoogle**
I added a simple use case of Storage that apple watch is able to load image successfully.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -204,9 +204,10 @@ participate in the Firebase community.
 
 ### tvOS, macOS, and Catalyst
```
Comment (paulb777) [inline README.md]: add watchOS to title

File: README.md [hunk 2]
```diff
@@ -204,9 +204,10 @@ participate in the Firebase community.
...
 Thanks to contributions from the community, many of Firebase SDKs now compile, run unit tests, and work on
-tvOS, macOS, and Catalyst.
+tvOS, macOS, watchOS and Catalyst.
 
 For tvOS, checkout the [Sample](Example/tvOSSample).
+For watchOS, checkout the [Independent Watch App Sample](Example/Messaging/App/watchOS/SampleWatchApp).
 
 Keep in mind that macOS, Catalyst and tvOS are not officially supported by Firebase, and this
 repository is actively developed primarily for iOS. While we can catch basic unit test issues with
```
Comment (paulb777) [inline README.md]: Add a comment for each pod listed below that does not yet have watchOS. Or a note at the top of the list.

**Comment by paulb777** [comment]
Please add a watchOS build test to the matrix in https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/storage.yml

**Comment by chliangGoogle** [comment]
I wonder if we need to bump up both FirebaseStorage and FirebaseAuthInterop's versions

File: README.md [hunk 3]
```diff
@@ -202,19 +202,20 @@ We've seen an amazing amount of interest and contributions to improve the Fireba
...
 Travis, there may be some changes where the SDK no longer works as expected on macOS or tvOS. If you
 encounter this, please [file an issue](https://github.com/firebase/firebase-ios-sdk/issues).
 
 During app setup in the console, you may get to a step that mentions something like "Checking if the app
-has communicated with our servers". This relies on Analytics and will not work on macOS/tvOS/Catalyst.
+has communicated with our servers". This relies on Analytics and will not work on macOS/tvOS/watchOS/Catalyst.
 **It's safe to ignore the message and continue**, the rest of the SDKs will work as expected.
 
 To install, add a subset of the following to the Podfile:
```
Comment (paulb777) [inline README.md]: Also the list below here needs to be clarified for watchOS

**Comment by paulb777** [comment]
It's some kind of test config problem.

In the log:

```
[!] CocoaPods could not find compatible versions for pod "FirebaseAuthInterop":
  In Podfile:
    FirebaseAuth (from `/Users/travis/build/firebase/firebase-ios-sdk/FirebaseAuth.podspec`) was resolved to 6.4.3, which depends on
      FirebaseAuthInterop (~> 1.0)
    FirebaseStorage (from `/Users/travis/build/firebase/firebase-ios-sdk/FirebaseStorage.podspec`) was resolved to 3.6.0, which depends on
      FirebaseAuthInterop (~> 1.1)
Specs satisfying the `FirebaseAuthInterop (~> 1.1), FirebaseAuthInterop (~> 1.0)` dependency were found, but they required a higher minimum deployment target.
```

I'll be able to help on this later this week

**Comment by chliangGoogle** [comment]
@doudounan is working on a fix of the pod lib lint error in watchOS introduced in #4944. It's strange it only got caught up in the Storage pod lib lint. @mikehaney24

**Comment by paulb777** [comment]
This is a different problem.   The Interop pods should be in the list here - https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/localize_podfile.swift#L27

File: README.md (hunk 2 above)
Comment (chliangGoogle) [reply README.md]: Done.

File: scripts/localize_podfile.swift [hunk 4]
```diff
@@ -26,7 +26,7 @@ let podfile = CommandLine.arguments[1]
 // versions when they're dependencies of other requested local pods.
 let implicitPods = ["FirebaseCore", "FirebaseInstanceID", "FirebaseInstallations", "Firebase",
                     "GoogleDataTransport", "GoogleDataTransportCCTSupport", "GoogleUtilities",
-                    "FirebaseCoreDiagnostics", "FirebaseRemoteConfig"]
+                    "FirebaseCoreDiagnostics", "FirebaseRemoteConfig", "FirebaseAuthInterop"]
```
Comment (paulb777) [inline scripts/localize_podfile.swift]: Thanks! All the interop pods should be added, but that can be done in a subsequent PR.

**Pull request merged**